managed/src/main/java/com/yugabyte/yw/common/CertificateHelper.java

```diff
@@ -151,109 +154,97 @@ public static UUID createRootCA(String nodePrefix, UUID customerUUID, String sto
+      clientCertSubject,
+      clientKeyPair.getPublic());
+    ContentSigner csrContentSigner = new JcaContentSignerBuilder("SHA256withRSA")
+      .build(pk);
+    PKCS10CertificationRequest csr = p10Builder.build(csrContentSigner);
+
+    KeyUsage keyUsage = new KeyUsage(KeyUsage.digitalSignature | KeyUsage.nonRepudiation |
+      KeyUsage.keyEncipherment | KeyUsage.keyCertSign);
+
+    X509v3CertificateBuilder clientCertBuilder = new X509v3CertificateBuilder(
+      subject, clientSerial, certStart, certExpiry,
+      csr.getSubject(), csr.getSubjectPublicKeyInfo());
+    JcaX509ExtensionUtils clientCertExtUtils = new JcaX509ExtensionUtils();
+    clientCertBuilder.addExtension(Extension.basicConstraints, true,
+      new BasicConstraints(false).toASN1Primitive());
+    clientCertBuilder.addExtension(Extension.authorityKeyIdentifier, false,
+      clientCertExtUtils.createAuthorityKeyIdentifier(cer));
+    clientCertBuilder.addExtension(Extension.subjectKeyIdentifier, false,
+      clientCertExtUtils.createSubjectKeyIdentifier(csr.getSubjectPublicKeyInfo()));
+    clientCertBuilder.addExtension(Extension.keyUsage, false, keyUsage.toASN1Primitive());
+
+    X509CertificateHolder clientCertHolder = clientCertBuilder.build(csrContentSigner);
+    X509Certificate clientCert = new JcaX509CertificateConverter()
+      .setProvider(new BouncyCastleProvider())
+      .getCertificate(clientCertHolder);
+
+    clientCert.verify(cer.getPublicKey(), "BC");
+
+    JcaPEMWriter clientCertWriter;
+    JcaPEMWriter clientKeyWriter;
+    StringWriter certWriter = new StringWriter();
+    StringWriter keyWriter = new StringWriter();
+    ObjectNode bodyJson = Json.newObject();
+    if (storagePath != null) {
+      String clientCertPath = String.format("%s/%s", storagePath, CLIENT_CERT);
+      String clientKeyPath = String.format("%s/%s", storagePath, CLIENT_KEY);
+      File clientCertfile = new File(clientCertPath);
+      File clientKeyfile = new File(clientKeyPath);
+      clientCertWriter = new JcaPEMWriter(new FileWriter(clientCertfile));
+      clientKeyWriter = new JcaPEMWriter(new FileWriter(clientKeyfile));
+    } else {
+      clientCertWriter = new JcaPEMWriter(certWriter);
+      clientKeyWriter = new JcaPEMWriter(keyWriter);
     }
+    clientCertWriter.writeObject(clientCert);
+    clientCertWriter.flush();
+    clientKeyWriter.writeObject(clientKeyPair.getPrivate());
+    clientKeyWriter.flush();
+    if (storagePath == null) {
+      bodyJson.put(CLIENT_CERT, certWriter.toString());
+      bodyJson.put(CLIENT_KEY, keyWriter.toString());
+    }
+    LOG.info("Created Client CA for username {} signed by root CA {}.", username, rootCA);
+    return bodyJson;
   }
 
   public static UUID uploadRootCA(
     String label, UUID customerUUID, String storagePath,
     String certContent, String keyContent, Date certStart,
     Date certExpiry, CertificateInfo.Type certType,
-    CertificateParams.CustomCertInfo customCertInfo) throws IOException {
```

Comment:
Also increase the scope of try (If required to cover whole method so even IOException gets caught and throw YWSE(INTERNAL_ERROR)  (Seee below comment)